### PR TITLE
update RTL styles

### LINKED ISSUE #465: Arabic/rtl style updates
:( !

image

This is similar to how the stepper looked in the storybook. Though it seems fixed there!

### PR DESCRIPTION
**Pull request by mark-prins**
Fixes #465 

Found more items, and I expect that there are others I've missed. This is enough for now though!

- [ ] Dashboard: spacing in StatusPanel
- [ ]  SidePanel: spacing around Actions and the action button + icons
- [ ]  IconWithButton: spacing between text and icon
- [ ]  Breadcrumb: spacing between text and icon
- [ ] Modal was not showing RTL at all
- [ ] Dialog buttons
- [ ] App footer: icon alignment
- [ ] padding in body of GeneralTab, most noticeable with the 'group by item' checkbox
- [ ] modal form label alignment

Dashboard
Screen Shot 2021-12-29 at 9 37 18 AM

Buttons
Screen Shot 2021-12-29 at 11 06 01 AM

General Tab
Screen Shot 2021-12-29 at 11 05 13 AM

Footer
Screen Shot 2021-12-29 at 11 04 53 AM

Detail Panel
Screen Shot 2021-12-29 at 9 36 31 AM

### CODE REVIEW

File: packages/common/src/ui/components/buttons/FlatButton.tsx [hunk 1]
```diff
@@ -10,27 +10,39 @@ interface ButtonProps {
...
-const StyledButton = styled(MuiButton)({
+const StyledButton = styled(MuiButton, {
+  shouldForwardProp: prop => prop !== 'isRtl',
+})<{ isRtl: boolean }>(({ isRtl }) => ({
   fontWeight: 700,
   marginLeft: 5,
   marginRight: 5,
   textTransform: 'none' as Property.TextTransform,
-});
+  '& .MuiButton-startIcon': isRtl
```
Comment (mark-prins) [inline packages/common/src/ui/components/buttons/FlatButton.tsx:20]: The MuiButton is adding an 'rtl' style but this has the same margins on it as the ltr version 😕 
am overriding manually with the values for left and right reversed

File: packages/common/src/ui/components/domain/Dashboard/StatsPanel.tsx [hunk 2]
```diff
@@ -30,7 +30,7 @@ export const StatsPanel: FC<StatsPanelProps> = ({
           color: 'gray.main',
           fontSize: '12px',
           fontWeight: 500,
-          marginLeft: '8px',
+          marginInlineStart: '8px',
```
Comment (mark-prins) [inline packages/common/src/ui/components/domain/Dashboard/StatsPanel.tsx:33]: `margin-inline-start` and `margin-inline-end` are new to me!
Not supported by IE 11 or some older browsers, but much neater to use than checking with `useRTL` everywhere! and the intention is clearer, so I think a good habit to use it.

File: packages/common/src/hooks/useDialog/useDialog.tsx [hunk 3]
```diff
@@ -42,16 +43,22 @@ enum Direction {
...
+    setSlide({
+      in: false,
+      direction: isRtl ? Direction.Right : Direction.Left,
+    });
     setTimeout(() => {
-      setSlide({ in: true, direction: Direction.Right });
+      setSlide({
+        in: true,
+        direction: isRtl ? Direction.Left : Direction.Right,
+      });
```
Comment (joshxg) [inline packages/common/src/hooks/useDialog/useDialog.tsx:61]: haha gotcha!

**Review by joshxg — APPROVED** [review]
Nice! Thanks heaps - good to keep on top of.

I'll contribute to the next list 🤣 

I see this fixed in your comment, so maybe just me? Seem to have a margin-left: -2px on the span for the 'start' icon, which if I remove, look better 🙄 
image

I'm confused about RTL :)

Curious.. should the breadcrumbs be 'round the other way' also?
image

Also wonder if we should fix this positioning.. looks OK though
image

I also wonder if we should change this. I'm thinking we should leave it in because it is a fun mini-game.. car racing! zoooom 🚗 🚗 

https://user-images.githubusercontent.com/35858975/147616173-b5611f8e-d6b3-4aaf-84c5-7550ccb6aca1.mp4

**Comment by mark-prins** [comment]
I get this on the dashboard.. don't know why it would be different for you!

Screen Shot 2021-12-29 at 5 18 41 PM

the snackbar! 🤣

**Comment by mark-prins** [comment]
Also not getting the snackbar madness 🤷  Though for me, the `X` icon isn't lining up so well..

Screen Shot 2021-12-29 at 5 45 31 PM

Fixed the breadcrumb - thanks! missed that one

**Pull request merged**